mux_test.go

```diff
@@ -1014,6 +1004,26 @@ func TestBuildVarsFunc(t *testing.T) {
 	}
 }
 
+// Addresses https://github.com/gorilla/mux/issues/200
+func TestCorrectVars(t *testing.T) {
+	tests := []routeTest{
+		{
+			title:        "Correct nested regex vars",
+			route:        new(Route).PathPrefix("/{id}/{kind:(a|b)}/{idx:[0-9]+}"),
```

Comment:
What about using a regexp replace to automatically turn capturing groups into non-capturing ones? Then we don't need a panic.